workflows/diagnostics/tests/offline/test_helpers.py

```diff
@@ -112,12 +115,27 @@ def test_insert_column_integrated_vars():
     xr.testing.assert_allclose(insert_column_integrated_vars(ds, ["Q1"]), expected)
 
 
+def test_load_grid_from_catalog(datadir):
+
+    res = "c12"
+    catalog_path = os.path.join(datadir, "catalog_dummy.yaml")
+
+    catalog = intake.open_catalog(catalog_path)
+
+    with pytest.raises(KeyError):
+        catalog[f"grid/{res}"].read()
```

Comment:
Sorry, I did not notice this before.  As written, this is not testing any of the core code that was modified in this PR.  I think it makes more sense to call `load_grid_info` under `pytest.raises` to test that the custom catalog it gets passed is being used.  It would be great if you could add a short comment describing this too.
```suggestion
    catalog_path = os.path.join(datadir, "catalog_dummy.yaml")

    with pytest.raises(KeyError):
        load_grid_info(res, catalog_path)
```